### PR TITLE
Imported Upstream version 3.1.10

### PR DESCRIPTION
**Pull request by stakoz**
TL;DR I have also added a trivial Zeroconf (aka Bonjour) service name support.

Given this configuration file:
```
; Global server settings
[Global]
    service name = MicroServer Gen8
    mimic model = Macmini4,1
```
One will see this in Finder and Zeroconf browsers instead of hostname:
```
$ avahi-browse -a
+ wlp2s0 IPv6 MicroServer Gen8                              Apple File Sharing   local
+ wlp2s0 IPv4 MicroServer Gen8                              Apple File Sharing   local
+ wlp2s0 IPv6 MicroServer Gen8                              Apple TimeMachine    local
+ wlp2s0 IPv4 MicroServer Gen8                              Apple TimeMachine    local
```
If the service name is missing the Netatalk will fallback to a host name as usual. In my case, this feature was crucial because, with the same box advertising both AFP and SMB services, Finder had a problem to distinguish what service type a user would like to use. Now AFP appears as Mac Mini with a human readable name and SMB is shown as a regular Windows box (one with a bluescreen).

Next I am going to discuss with Netatalk maintainers if it is feasible to add an official support (including updating man pages, documentation, etc.) for specifying the service name. Please let me know what do you think. Thank you in advance.

### CODE REVIEW

**Comment by rossburton** [comment]
I was just looking for a 3.1.10 upgrade, this is great.  I don't suppose you've published debs for jessie? :)

**Comment by adiknoth** [comment]
Hi!

Thanks for the pull request. Two things:

(1) Please always separate upstream imports from any other modifications. That is, this PR should have been at least three individual commits: first commit from

   $ gbp import-orig --sign-tags /path/to/netatalk-3.1.10.tar.bz2

  then individual commits for any Debian changes, one change at a time, and then your separate patch for zeroconf.

(2) Your zeroconf patch goes into debian/patch/XXX_zeroconf.patch. Use the quilt tool for that.

Realistically, I'd prefer this patch being applied upstream and then imported as part of 3.1.11. Good that you're already talking to upstream, thanks for that!

Can I ask you a favour? We have two patches in Debian that could potentially go into 3.1.11, so that we no longer have to keep them around:

   https://github.com/adiknoth/netatalk-debian/blob/master/debian/patches/402_init_lsb.patch
   https://github.com/adiknoth/netatalk-debian/blob/master/debian/patches/114_fix_macusers_ps_parsing.patch

Since you're already talking to upstream, could you please ask them to integrate these two? Thanks!

Last not least: I've updated my repo to 3.1.10, so this PR no longer cleanly applies. If you want, rebase (realistically: start over) and put your zeroconf patch into debian/patches/, then enable it in debian/patches/series. As said before, I'd rather wait and get this change via 3.1.11, but if the patch is trivial, I'm happy to ship it in here until upstream releases it.



Thanks!

**Comment by stakoz** [comment]
Hello Adrian.

Thanks for your detailed review and workflow hints. To be honest, I completely forgot about notifying Netatalk maintainers. Thank you for reminding me about it. :-) I sent the email immediately after I've received your message. Let's wait and see. If I get any response I am going to see what can be done about the patches you mentioned.

Ross, I am sorry I don't have binaries at the moment, but I do have a Jessie PBuilder environment. Drop me a note and I can build one for you.

--
Stan
